File: frontend/src/services/pullRequests.ts
import { apiFetchSafe } from "@/services/api";

export interface PullRequestItem {
  id: string;
  ticketId: string;
  title: string;
  status: string;
  confidence: number;
  severity?: string;
  createdAt?: string | null;
}

export async function getPullRequests(): Promise<PullRequestItem[]> {
  return apiFetchSafe("/pull-requests", []);
}

File: backend/api/routes/pull_requests.py
"""Pull request API routes."""

from fastapi import APIRouter
from services.run_store import list_runs

router = APIRouter(tags=["pull_requests"])


@router.get("/pull-requests")
async def list_pull_requests():
    runs = list_runs()
    prs = []
    for run in runs:
        prs.append({
            "id": f"pr-{run.ticket_id}",
            "ticketId": run.ticket_id,
            "title": f"fix: {run.classification}",
            "status": "open" if run.status.value != "failed" else "draft",
            "confidence": run.confidence,
            "severity": run.severity,
            "createdAt": run.completed_at,
        })
    return prs
